mongo_connector/doc_managers/elastic_doc_manager.py

```diff
@@ -118,11 +136,23 @@ def update(self, document_id, update_spec, namespace, timestamp):
         """
         self.commit()
         index, doc_type = self._index_and_mapping(namespace)
-        document = self.elastic.get(index=index, doc_type=doc_type,
-                                    id=u(document_id))
-        updated = self.apply_update(document['_source'], update_spec)
-        # _id is immutable in MongoDB, so won't have changed in update
-        updated['_id'] = document['_id']
+
+        document = None
+        if doc_type in self.routing and 'parentPath' in self.routing[doc_type]:
+            # we need to manually retrieve the parent id from mongo
+            # this is due to the fact that elasticsearch needs the parent id to
+            # know where to route the get request; we might not have the parent
+            # id available in our update request though
+            db, collection = self.command_helper.revert_namespace(namespace)
+            updated = self.mongo[db][collection].find_one(document_id)
```

Comment:
I see, good explanation. However, I'm not sure that it's necessary to treat documents differently here if they have a parent/routing. Won't the document as stored in Elasticsearch (and retrieved by `get()` below) also have this information?